src/Security/SecurityControllerBuilder.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Security;
+
+use Symfony\Bundle\MakerBundle\Util\ClassSourceManipulator;
+
+/**
+ * @internal
+ */
+final class SecurityControllerBuilder
+{
+    public function addLoginMethod(ClassSourceManipulator $manipulator)
+    {
+        $loginMethodBuilder = $manipulator->createMethodBuilder('login', 'Response', false, ['@Route("/login", name="app_login")']);
+        $loginMethodBuilder->addParam(
+            (new \PhpParser\Builder\Param('authenticationUtils'))->setTypeHint('AuthenticationUtils')
+        );
+
+        $manipulator->addMethodBody($loginMethodBuilder, <<<'CODE'
+<?php
+// get the login error if there is one
+$error = $authenticationUtils->getLastAuthenticationError();
+// last username entered by the user
+$lastUsername = $authenticationUtils->getLastUsername();
+CODE
+        );
+        $loginMethodBuilder->addStmt($manipulator->createMethodLevelBlankLine());
+        $manipulator->addMethodBody($loginMethodBuilder, <<<'CODE'
+<?php
+return $this->render(
+    'security/login.html.twig',
+    [
+        'last_username' => $lastUsername,
+        'error' => $error,
+    ]
+);
+CODE
+        );
+        $manipulator->addMethodBuilder($loginMethodBuilder);
+        $manipulator->addUseStatementIfNecessary('Symfony\\Component\\HttpFoundation\\Response');
+        $manipulator->addUseStatementIfNecessary('Symfony\\Component\\Routing\\Annotation\\Route');
+        $manipulator->addUseStatementIfNecessary('Symfony\\Component\\Security\\Http\\Authentication\\AuthenticationUtils');
```

Comment:
Can we move these `addUseStatementIfNecessary` up a little? It's a nit-pick, but I think it's clearer if they are right near where they are used/needed.

Also, we can use the `::class` syntax with `use` statements on top.